src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/IAnalyzeResultWriter.cs

```diff
@@ -11,6 +11,6 @@ namespace Microsoft.DotNet.UpgradeAssistant.Analysis
 {
     public interface IAnalyzeResultWriter
     {
-        Task WriteAsync(IAsyncEnumerable<AnalyzeResultDefinition> results, CancellationToken token);
+        Task WriteAsync(IAsyncEnumerable<AnalyzeResultDefinition> results, string? format, CancellationToken token);
```

Comment:
It would be great to have a provider that would give you the correct writer given a format so we can separate out the writing from the format selection. For example:

```csharp
public interface IAnalyzeResultWriterProvider
{
  bool TryGetWriter(string? format, out IAnalyzeResultWriter writer);
}
```

Reply:
No need to block for now on this - I'll create a new issue to track doing that.

Reply:
see https://github.com/dotnet/upgrade-assistant/issues/976